datafusion/proto/proto/datafusion.proto

```diff
@@ -714,6 +714,9 @@ message Union{
 }
 
 message ScalarListValue{
+    // encode null explicitly to distinguish a list with a null value
+    // from a list with no values)
+    bool is_null = 3;
```

Comment:
I hope to remove this when I remove `PrimitiveScalarType` and use `ArrowType` instead